notes.md

```diff
@@ -0,0 +1,124 @@
+
+  - Authorized, but unissued shares can be held in an address owned by the company.
+  - We do have to differentiate between never issued and repurchased tokens (treasury shares), however.
+
+      Perhaps just another multisig wallet.
+
+  - Price per share is denominated in USD.
+  - CO2KN, INC will announce its own exchange rate at some point close to the auction.
+  - So, no need for an oracle. This is optimal because the IRS sees each transaction of ETH as a sale. So it's not viewed as one transaction of ETH for Equity, it's seen as USD for equity and then cash for ETH.
+  - Nature of securities takes the following values: `equity`, `debt`, `convertible`, etc.
+  - 'Closing soon' can probably be posted a week before close.
+
+## also need to record
+
+* Name of the shareholder;
+* Complete mailing address of the stock shareholder including contact number;
+* Stock certificate number;
+* The total number of shares outstanding;
+* The date the shares were purchased;
+* How much the person paid per share (consideration/monetary value);
+* The class of shares;
+* Why the shares were transferred.
+
+### implementation
+
+* The `shareholder name`, their `mailing address`, and `contact number` MUST all be stored in an off-chain database and the content hashed, and the `hash` associated with the address.
+* The total number of shares outstanding is the difference between a cap and `token.totalSupply()` and is rightly a function of the `CrowdsaleContract`.
+* The date the share was purchased is a property of the block the purchase was processed in so there is no need to store it in the token. However the user database might wish to record that information as well for convenience.
+* The price the user paid is a function of the ETH price per Token (which can change during the crowdsale) and ETH price at that point (which can also be set in the `CrowdsaleContract`) and the number of ETH spent, which the `CrowdsaleContract` also knows and which is returned in the transaction receipt.
+* With regard to the certificate number, the token should record the certificate numbers, either algorithmically if possible, or else it will need to be generated offchain and imported in bulk when the certificates themselves are issued.
+* A certificate might represent more than one share, in which case it will need to record a range of inputs in the form of an encoded bytes array.
+* as shares are traded a script needs to pull Transfer events from the global blockchain and keep the off-chain records up to date, either in real-time, or in an ad-hoc way.  This should be the job of the Crowdsale Admin website.
+
+### Questions and Scenarios.
+
+1. When someone buys a Token does that mean they have bought a specific Stock with a specific certificate number, or does it mean they've bought the right to be issued a stock certificate?
+
+    scenario: Token Buys Share immediately.
+    scenario: Token Buys rights to a share, when the crowdsale is finished.
```

Comment:
I think we should differentiate between tokens during the crowdsale and releasable tokens so I'd imagine its a right to a share and becomes a share after the crowdsale similar to this: https://github.com/TokenMarketNet/ico/blob/master/contracts/ReleasableToken.sol

Reply:
Discussed this with @harrisonperl who prefers we just release tokens (one token == one share) on a first-come, first-served basis.

Reply:
Will discuss with Jeff tomorrow, but yes I prefer if we can just release tokens. I think that is possible so long as we include a clause that the round will close above $0.